e2e/cypress/integration/notifications/ignore_channel_mentions.js

```diff
@@ -0,0 +1,132 @@
+    // # Then click it to access the drop-down menu
+    cy.get('#channelHeaderTitle').click();
+
+    // * The dropdown menu of the channel header should be visible;
+    cy.get('#channelLeaveChannel').should('be.visible');
+
+    // # Click 'Add Members'
+    cy.get('#channelAddMembers').click();
+
+    // * Assert that modal appears
+    // # Click the first row for a number of times
+    Cypress._.times(num, () => {
+        cy.get('#multiSelectList').should('be.visible').children().first().click();
+    });
+
+    // # Click the button "Add" to add user to a channel
+    cy.get('#saveItems').click();
+
+    // # Wait for the modal to disappear
+    cy.get('#addUsersToChannelModal').should('not.exist');
+}
+
+function setIgnoreMentions(toSet) {
+    // # Click on the header to open dropdown
+    cy.get('#channelHeaderDropdownButton').should('exist').click();
+
+    // # Click on the notification preferences
+    cy.get('#channelNotificationPreferences').should('exist').click();
+
+    // # Click on the edit button for ignore channel mentions
+    cy.get('#ignoreChannelMentionsEdit').should('exist').click();
+
+    // # Click on selected option
+    cy.get(`#ignoreChannelMentions${toSet}`).should('exist').click();
+
+    // # Click on save to save the configuration
+    cy.get('#saveSetting').should('exist').click();
+
+    // * Assert that the option selected is reflected
+    cy.get('#ignoreChannelMentionsDesc').should('contain', toSet);
+
+    // # Click on the X button to close the modal
+    cy.get('#channelNotificationModalLabel').siblings('.close').click();
+}
+
+describe('CS15445 Join/leave messages', () => {
+    let testTeam;
+    let userB;
+
+    let channelA;
+    let channelB;
+
+    before(() => {
+        // # Login as new user and visit town-square
+        cy.apiInitSetup().then(({team, user}) => {
+            testTeam = team;
+
+            // # Add 1 users
+            cy.apiCreateUser().then(({user: newUser}) => { // eslint-disable-line
+                userB = newUser;
+                cy.apiAddUserToTeam(testTeam.id, userB.id);
+            });
+
+            // # Create two channels
+            cy.apiCreateChannel(testTeam.id, 'channel-test', 'Channel').then((res) => {
+                channelA = res.body;
+            });
+            cy.apiCreateChannel(testTeam.id, 'channel-test', 'Channel').then((res) => {
+                channelB = res.body;
+            });
+
+            cy.apiLogin(user);
+        });
+    });
+
+    it('MM-T567 - Turn on Ignore mentions for @channel, @here and @all', () => {
+        cy.visit(`/${testTeam.name}/channels/${channelA.name}`);
+
+        // # Add users to channel
+        addNumberOfUsersToChannel(1);
+
+        cy.getLastPostId().then((id) => {
+            // * The system message should contain 'added to the channel by you'
+            cy.get(`#postMessageText_${id}`).should('contain', 'added to the channel by you');
+        });
+
+        setIgnoreMentions('On');
```

Comment:
I think the convention is to add a description comment here, as in 
```suggestion
        // # Set Ignore mentions setting to On
        setIgnoreMentions('On');
```